src/main/java/ev3dev/actuators/ev3/EV3Led.java

```diff
@@ -21,16 +24,26 @@
         LEFT,
         RIGHT
     }
+    
+    private static final Direction[] directionArray = {Direction.LEFT,Direction.RIGHT};
 
     private static final Logger log = LoggerFactory.getLogger(EV3Led.class);
 
+    /**
+     * @deprecated Use EV3LedDirection.LEFT instead.
+     */
+    @Deprecated
     public static final int LEFT = 0;
+    /**
+     * @deprecated Use EV3Led.Direction.RIGHT instead.
+     */
+    @Deprecated
     public static final int RIGHT = 1;
 
     private final Direction direction;
 
-    private final String LED_RED;
-    private final String LED_GREEN;
+    private final DataChannelRewriter redWriter;
+    private final DataChannelRewriter greenWriter;
```

Comment:
I realized here that some attributes may not be read-only or write-only. For example, the `duty_cycle_sp` attribute is both written and read in `BaseRegulatedMotor`. This requires having two separate objects for the reads & writes (that might be OK though).

Reply:
I've only prototyped BaseRegulatedMotor so far. Are there any conditions where we'd want read and write in the same interaction? Or where they'd interfere with each other? We could build something like a random access file if needed.

Reply:
IIUC they will not interfere with each other, so having two objects, one for reading and the other for writing, should work fine. I just got surprised that we are going to have two separate objects for one "real" object/sysfs attribute.